src/main/java/seedu/address/logic/parser/SortOrder.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents the sorting order of a sort command.
+ */
+public class SortOrder {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Sort order should be 'a' for ascending or 'd' for descending, and it should not be blank";
+
+    /**
+     * Sort order must be 'a' or 'd'.
+     */
+    public static final String VALIDATION_REGEX = "^[ad]$";
+    private final boolean isSortByAscending;
+
+    /**
+     * Creates a SortOrder object with the corresponding sorting order.
+     * @param sortOrder Represents ascending with 'a' and descending with 'd'.
+     */
+    public SortOrder(String sortOrder) {
+        requireNonNull(sortOrder);
+        checkArgument(isValidSortOrder(sortOrder), MESSAGE_CONSTRAINTS);
+        this.isSortByAscending = sortOrder.equals("a");
```

Comment:
I think the sort order should be changed into a switch case in order to test properly for correct input and so we can practise defensive programming